deflate_rle.c

```diff
@@ -58,7 +58,9 @@ Z_INTERNAL block_state deflate_rle(deflate_state *s, int flush) {
 
         /* Emit match if have run of STD_MIN_MATCH or longer, else emit literal */
         if (match_len >= STD_MIN_MATCH) {
-            check_match(s, s->strstart, s->strstart - 1, match_len);
+            Assert((s->strstart - 1) <= UINT16_MAX, "strstart-1 should fit in uint16_t");
+            Assert(s->match_start <= UINT16_MAX, "match_start should fit in uint16_t");
+            check_match(s, (Pos)s->strstart, (Pos)(s->strstart - 1), match_len);
```

Comment:
No `match_start` here, only `strstart` needs to fit in uint16_t.

Reply:
Right, and one of the asserts is not needed. Fixing it now.